rfcs/text/0027-faas-fields.md

```diff
@@ -52,6 +54,35 @@ Stage X: Provide a brief explanation of why the proposal is being marked as aban
 Stage 1: Describe at a high level how this change affects fields. Include new or updated yml field definitions for all of the essential fields in this draft. While not exhaustive, the fields documented here should be comprehensive enough to deeply evaluate the technical considerations of this change. The goal here is to validate the technical details for all essential fields and to provide a basis for adding experimental field definitions to the schema. Use GitHub code blocks with yml syntax formatting, and add them to the corresponding RFC folder.
 -->
 
+Discussing the initial proposal with Andrew Wilkins, we came up with an adapted proposal (compared to the proposal for stage 0) that would reuse as many as possible existing ECS fields:
+
+### New Fields
+Field | Type | Example | Description | Use case
+ -- | -- | --  | --  |--
+faas.coldstart | boolean | true | Boolean value indicating a cold start of a function | Can be used in the UI denote function coldstarts.
+faas.execution | keyword | "af9d5aa4-a685-4c5f-a22b-444f80b3cc28" | The execution ID of the current function execution. | Allows correlation with CloudWatch logs and metrics
+faas.trigger.type | keyword | "http" | one of `http`,`pubsub`,`datasource`, `timer`, `other` | Allows differentiating different function types
+faas.trigger.request_id | keyword | e.g. `123456789` | The iD of the trigger request , message, event, etc. | Correlation of metrics and logs with the corresponding trigger request
+
+### Reusing existing `service.*` fields
+For the initially proposed fields `faas.name`, `faas.id`, `faas.version` and `faas.instance` we decided to reuse the existing fields `service.name`, `service.id`, `service.version` and `service.node.name`.
+
+### Nesting `cloud.*` and `service.*` fields under `_.origin.*` and `_.target.*`
+We identified a big overlap between the initially proposed `faas.trigger.*` fields with the already existing `cloud.*` and `service.` fields. 
+Allowing to **self-nest cloud and service fields** under `cloud.origin.*` / `cloud.target.*` and `service.origin.*` / `service.target.*`, respectively, would allow to cover most of the `faas.trigger.*` fields.
```

Comment:
I propose instead of nesting at `_.origin.*`, the existing top-level field set represents the origin. The target would be nested at `_.target.*` as proposed.

This approach continues the pattern adopted for the user field set: `user.*` and `user.target.*`. The user/service/entity performing the main action is captured under the top-level with the user/service/entity affected by the action residing at `_.target.*`.

Example :

```json
{
  "service": {
    "name": "origin-service",
    "target": {
      "name": "target-service"
    }
  }
}
```

Using the existing top-level fields for the event "doer" lets users continue using existing queries. For example, only querying `service.name` instead of needing to account for both `service.name` and `service.origin.name`.

Reply:
`origin` is in addition to `service.name`, it gives more context as to where the function was invoked from. 

```
CALLEE -> FUNCTION INVOCATION -> OUTGOING CALL
```

the proposal allows us to set information for all three stages:

```
service.origin.name -> service.name -> service.target.name
```

With `service.*` always being present and relating the the `do'er` or `FUNCTION INVOCATION` itself.

Reply:
Thanks, @Mpdreamz, for the follow-up. I also reviewed the [trigger examples](https://github.com/elastic/apm/blob/b98e151e4379f17754a393cb080875f0ca36e1b4/specs/agents/tracing-instrumentation-aws-lambda.md#trigger-specific-instrumentation) from elastic/apm#470, and those examples also helped me to understand these FaaS use cases better.

I'm still hesitant that introducing the `_.origin.*` nestings could create confusion. For example, when a user should select `service.name` or `cloud.service.name` versus `service.origin.name` or `cloud.origin.service.name`. However, if we define the difference clearly in the ECS docs, we can hopefully avoid confusion between the two.

I have no objection to moving forward with the `_.origin.*` nestings as proposed. However, can we please summarize what we discussed here as a potential concern in the `Concerns` section?